pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -110,19 +110,23 @@ func (a *cpuAccumulator) freeCPUs() []int {
 			// as each core.
 			iSocketColoScore := a.topo.CPUDetails.CPUsInSocket(iSocket).Intersection(a.result).Size()
 			jSocketColoScore := a.topo.CPUDetails.CPUsInSocket(jSocket).Intersection(a.result).Size()
+			if iSocketColoScore > jSocketColoScore {
+				return true
+			}
 
 			// Compute the number of available CPUs available on the same socket
 			// as each core.
 			iSocketFreeScore := a.details.CPUsInSocket(iSocket).Size()
 			jSocketFreeScore := a.details.CPUsInSocket(jSocket).Size()
+			if iSocketFreeScore < jSocketFreeScore {
+				return true
+			}
 
 			// Compute the number of available CPUs on each core.
 			iCoreFreeScore := a.details.CPUsInCore(iCore).Size()
 			jCoreFreeScore := a.details.CPUsInCore(jCore).Size()
 
-			return iSocketColoScore > jSocketColoScore ||
-				iSocketFreeScore < jSocketFreeScore ||
-				iCoreFreeScore < jCoreFreeScore ||
+			return iCoreFreeScore < jCoreFreeScore ||
 				iSocket < jSocket ||
```

Comment:
Can we move `iSocket < jSocket` and `iCore < jCore` to underneath their variable definition as well?

Reply:
The intention is to keep the priority of iCoreFreeScore < jCoreFreeScore check over the socket check.
That was why the 3 checks are left in the same order as before refactoring.